Quaver.Shared/Screens/Select/SelectScreen.cs

```diff
@@ -318,6 +321,129 @@ private static void HandleKeyPressTab()
             else if (ConfigManager.LeaderboardSection.Value == LeaderboardType.Global)
                 ConfigManager.LeaderboardSection.Value = LeaderboardType.Local;
         }
+        
+        /// <summary>
+        ///     Handles when the user presses the del key
+        /// </summary>
+        private void HandleKeyPressDel()
+        {
+            var view = View as SelectScreenView;
+
+            if (!KeyboardManager.IsUniqueKeyPress(Keys.Delete))
+                return;
+
+            if (DialogManager.Dialogs.Count > 0)
+                return;
+
+            try
+            {
+                var selectedMapsetIndex = view.MapsetScrollContainer.SelectedMapsetIndex;
+                var selectedDifficultyIndex = view.DifficultyScrollContainer.SelectedMapIndex;
+                var selectedMapset = AvailableMapsets[selectedMapsetIndex];
+                var selectedDifficulty = selectedMapset.Maps[selectedDifficultyIndex];
+                var selectedMapsetPath = Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory);
+                var selectedDifficultyPath = Path.Combine(selectedMapsetPath, selectedDifficulty.Path);
+
+                ConfirmDialog confirmDelete = null;
+
+                switch (view.ActiveContainer)
+                {
+                    case SelectContainerStatus.Mapsets:
+                        confirmDelete = new ConfirmDialog("Are you sure you want to delete this mapset?");
+
+                        confirmDelete.Confirmation += (sender, confirm) =>
+                        {
+                            if (confirm)
+                            {
+                                if (AvailableMapsets.Count == 0)
+                                    return;
+
+                                Logger.Log($"Deleting Mapset \"{selectedMapset.Artist} - {selectedMapset.Title}\"",
+                                    LogLevel.Important, LogType.Runtime);
+
+                                // Dispose of the currently playing track, to release the resource from it's stream.
+                                AudioEngine.Track.Dispose();
```

Comment:
`AudioEngine.Track?.Dispose()` instead. You need to handle the case of where a track might not be loaded.